pkg/services/alerting/engine.go

```diff
@@ -16,11 +16,25 @@ import (
 	"github.com/grafana/grafana/pkg/infra/usagestats"
 	"github.com/grafana/grafana/pkg/models"
 	"github.com/grafana/grafana/pkg/services/encryption"
+	"github.com/grafana/grafana/pkg/services/notifications"
 	"github.com/grafana/grafana/pkg/services/rendering"
 	"github.com/grafana/grafana/pkg/setting"
 	"github.com/grafana/grafana/pkg/tsdb/legacydata"
 )
 
+// AlertStore is a subset of SQLStore API to satisfy the needs of the alerting service.
+// A subset is needed to make it easier to mock during the tests.
+type AlertStore interface {
+	GetAllAlertQueryHandler(context.Context, *models.GetAllAlertsQuery) error
+	GetDataSource(context.Context, *models.GetDataSourceQuery) error
+	GetDashboardUIDById(context.Context, *models.GetDashboardRefByIdQuery) error
```

Comment:
why are the 2 methods above included in the AlertStore interface?

Reply:
AlertStore is _not_ a store of Alerts, but rather a subset of the Store interface _used_ by the alerting service for all its needs. The methods above are all used by the old alerting internally, thus - they are part of this interface.